src/test/java/com/google/googleidentity/filter/UserAuthenticationFilterTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+    Copyright 2020 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package com.google.googleidentity.filter;
+
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.googleidentity.security.UserSession;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertEquals;
+
+import com.google.inject.Provider;
+
+import org.junit.Test;
+
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Test {@link UserAuthenticationFilter}'s logic
+ */
+public class UserAuthenticationFilterTest {
+
+    private UserSession session= new UserSession();
+
+    public class UserSessionProvider1 implements Provider<UserSession> {
+        @Override
+        public UserSession get() {
+            return session;
+        }
+    }
```

Comment:
No need for this. You can just use Providers.of(session) in the test.

https://google.github.io/guice/api-docs/latest/javadoc/com/google/inject/util/Providers.html#of-T-